xarray/test/test_dataarray.py

```diff
@@ -1791,3 +1791,41 @@ def test_full_like(self):
         actual = _full_like(DataArray([1, 2, 3]), fill_value=np.nan)
         self.assertEqual(actual.dtype, np.float)
         np.testing.assert_equal(actual.values, np.nan)
+    
+    def test_dot(self):
+        x = np.linspace(-3, 3, 6)
+        y = np.linspace(-3, 3, 5)
+        z = range(4) 
+        da_vals = np.arange(6 * 5 * 4).reshape((6, 5, 4))
+        da = DataArray(da_vals, coords=[x, y, z], dims=['x', 'y', 'z'])
+        
+        dm_vals = range(4)
+        dm = DataArray(dm_vals, coords=[z], dims=['z'])
+        
+        #nd dot 1d
```

Comment:
style nit: comments should start with a single space:
https://www.python.org/dev/peps/pep-0008/#block-comments